samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/api/controller/ShoppingCartsController.java

```diff
@@ -0,0 +1,176 @@
+import io.eventdriven.ecommerce.shoppingcarts.gettingbyid.ShoppingCartDetails;
+import io.eventdriven.ecommerce.shoppingcarts.gettingcarts.GetShoppingCarts;
+import io.eventdriven.ecommerce.shoppingcarts.gettingcarts.ShoppingCartShortInfo;
+import io.eventdriven.ecommerce.shoppingcarts.opening.OpenShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.productitems.PricedProductItem;
+import io.eventdriven.ecommerce.shoppingcarts.productitems.ProductItem;
+import io.eventdriven.ecommerce.shoppingcarts.removingproductitem.RemoveProductItemFromShoppingCart;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.media.Schema;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartsController {
+  private final CommandHandler<OpenShoppingCart> handleInitializeShoppingCart;
+  private final CommandHandler<AddProductItemToShoppingCart> handleAddProductItemToShoppingCart;
+  private final CommandHandler<RemoveProductItemFromShoppingCart> handleRemoveProductItemFromShoppingCart;
+  private final CommandHandler<ConfirmShoppingCart> handleConfirmShoppingCart;
+  private final CommandHandler<CancelShoppingCart> handleCancelShoppingCart;
+  private final QueryHandler<GetShoppingCartById, Optional<ShoppingCartDetails>> handleGetShoppingCartById;
+  private final QueryHandler<GetShoppingCarts, Page<ShoppingCartShortInfo>> handleGetShoppingCarts;
+
+  public ShoppingCartsController(
+    CommandHandler<OpenShoppingCart> handleInitializeShoppingCart,
+    CommandHandler<AddProductItemToShoppingCart> handleAddProductItemToShoppingCart,
+    CommandHandler<RemoveProductItemFromShoppingCart> handleRemoveProductItemFromShoppingCart,
+    CommandHandler<ConfirmShoppingCart> handleConfirmShoppingCart,
+    CommandHandler<CancelShoppingCart> handleCancelShoppingCart,
+    QueryHandler<GetShoppingCartById, Optional<ShoppingCartDetails>> handleGetShoppingCartById,
+    QueryHandler<GetShoppingCarts, Page<ShoppingCartShortInfo>> handleGetShoppingCarts) {
+
+    this.handleInitializeShoppingCart = handleInitializeShoppingCart;
+    this.handleAddProductItemToShoppingCart = handleAddProductItemToShoppingCart;
+    this.handleRemoveProductItemFromShoppingCart = handleRemoveProductItemFromShoppingCart;
+    this.handleConfirmShoppingCart = handleConfirmShoppingCart;
+    this.handleCancelShoppingCart = handleCancelShoppingCart;
+    this.handleGetShoppingCartById = handleGetShoppingCartById;
+    this.handleGetShoppingCarts = handleGetShoppingCarts;
+  }
+
+  @PostMapping
```

Comment:
@PostMapping should proabably returns 201 CREATED.
You can do that by additional annotation:
`@ResponseStatus(HttpStatus.CREATED)`

Reply:
Fixed in https://github.com/oskardudycz/EventSourcing.JVM/pull/1/commits/ec0198125a019daa9bdfdc574e7cc51ef7761d6e.